datafusion/spark/src/function/string/concat.rs

```diff
@@ -122,13 +127,13 @@ fn spark_concat(args: ScalarFunctionArgs) -> Result<ColumnarValue> {
     apply_null_mask(result, null_mask)
 }
 
-/// Compute NULL mask for the arguments
-/// Returns None if all scalars and any is NULL, or a Vector of
-/// boolean representing the null mask for incoming arrays
+/// Compute NULL mask for the arguments using NullBuffer::union
+/// Returns None if all scalars and any is NULL, or an Option of NullBuffer
+/// representing the combined null mask for incoming arrays
 fn compute_null_mask(
     args: &[ColumnarValue],
     number_rows: usize,
-) -> Result<Option<Vec<bool>>> {
+) -> Result<Option<Option<NullBuffer>>> {
```

Comment:
Thoughts on defining a separate enum here instead of encoding meaning in a double Option?

```rust
enum NullMask {
    AllScalarsWithNull,
    AllScalarsNoNulls,
    Nulls(Option<NullBuffer>),
}
```

Reply:
Thanks @Jefffrey I agree, originally I thought introducing another enum just for this function would be too much, but I think it would be better having enum